File: src/plugin/inter/GenieSkillSelectionInterfacePlugin.java
package plugin.inter;

import org.redrune.game.content.plugin.type.InterfacePlugin;
import org.redrune.game.entity.actor.player.Player;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 2019-01-29
 */
public class GenieSkillSelectionInterfacePlugin implements InterfacePlugin {
	
	/**
	 * The interface id for the skill selection interface
	 */
	private static final int SKILL_SELECTION_INTERFACE_ID = 1139;
	
	@Override
	public boolean handle(Player player, int interfaceId, int componentId, int itemId, int slotId, int packetId) {
		return true;
	}
	
	@Override
	public void register() {
		registerInterfacePlugin(SKILL_SELECTION_INTERFACE_ID);
	}
	
	public static void displayInterface(Player player) {
		player.getVarManager().sendVar(261, 0);
		player.getInterfaceManager().sendInterface(SKILL_SELECTION_INTERFACE_ID);
	}
}

File: src/main/java/plugin/rsinterface/GenieSkillSelectionInterfacePlugin.kt
package plugin.rsinterface

import game.content.plugin.type.InterfacePlugin
import game.entity.actor.player.Player

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 2019-01-29
 */
class GenieSkillSelectionInterfacePlugin : InterfacePlugin {
    override fun handle(
        player: Player,
        interfaceId: Int,
        componentId: Int,
        itemId: Int,
        slotId: Int,
        packetId: Int
    ): Boolean {
        return true
    }

    override fun register() {
        registerInterfacePlugin(SKILL_SELECTION_INTERFACE_ID)
    }

    companion object {
        /**
         * The interface id for the skill selection interface
         */
        private const val SKILL_SELECTION_INTERFACE_ID = 1139
        fun displayInterface(player: Player) {
            player.varManager.sendVar(261, 0)
            player.interfaceManager.sendInterface(SKILL_SELECTION_INTERFACE_ID)
        }
    }
}